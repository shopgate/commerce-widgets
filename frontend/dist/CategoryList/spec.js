'use strict';

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _jsxFileName = 'frontend/src/CategoryList/spec.jsx'; /**
                                                          * Copyright (c) 2017, Shopgate, Inc. All rights reserved.
                                                          *
                                                          * This source code is licensed under the Apache 2.0 license found in the
                                                          * LICENSE file in the root directory of this source tree.
                                                          */

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _enzyme = require('enzyme');

var _List = require('../../../../../components/List');

var _List2 = _interopRequireDefault(_List);

var _index = require('./index');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

describe('<CategoryListWidget />', function () {
  it('should not render the CategoryListWidget', function () {
    var props = {
      fetchCategory: function fetchCategory() {},
      items: null,
      settings: {
        categoryNumber: '',
        headline: 'Yay Categories',
        showImages: false
      }
    };

    var wrapper = (0, _enzyme.mount)(_react2.default.createElement(_index.CategoryListWidget, _extends({}, props, {
      __source: {
        fileName: _jsxFileName,
        lineNumber: 26
      }
    })));

    expect(wrapper.find(_List2.default.Item).length).toBe(0);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render the CategoryListWidget', function () {
    var props = {
      fetchCategory: function fetchCategory() {},
      items: [{
        id: '1',
        name: 'Headline',
        imageUrl: '/some/url'
      }, {
        id: '2',
        name: 'Headline',
        imageUrl: '/some/url'
      }],
      settings: {
        categoryNumber: '',
        headline: 'Yay Categories',
        showImages: false
      }
    };

    var wrapper = (0, _enzyme.mount)(_react2.default.createElement(_index.CategoryListWidget, _extends({}, props, {
      __source: {
        fileName: _jsxFileName,
        lineNumber: 56
      }
    })));

    expect(wrapper.find(_List2.default.Item).length).toBe(2);
    expect(wrapper).toMatchSnapshot();
  });
});
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }
}();

;