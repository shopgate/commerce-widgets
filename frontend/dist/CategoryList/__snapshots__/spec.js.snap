// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CategoryListWidget /> should not render the CategoryListWidget 1`] = `
<CategoryListWidget
  fetchCategory={[Function]}
  getCategory={[Function]}
  items={null}
  settings={
    Object {
      "categoryNumber": "",
      "headline": "Yay Categories",
      "showImages": false,
    }
  }
/>
`;

exports[`<CategoryListWidget /> should render the CategoryListWidget 1`] = `
<CategoryListWidget
  fetchCategory={[Function]}
  getCategory={[Function]}
  items={
    Array [
      Object {
        "id": "1",
        "imageUrl": "/some/url",
        "name": "Headline",
      },
      Object {
        "id": "2",
        "imageUrl": "/some/url",
        "name": "Headline",
      },
    ]
  }
  settings={
    Object {
      "categoryNumber": "",
      "headline": "Yay Categories",
      "showImages": false,
    }
  }
>
  <div
    className="css-1tgouoh"
  >
    <h3
      className="css-twewb4"
    >
      Yay Categories
    </h3>
    <List>
      <List
        className=""
      >
        <ul
          className="list-style-none m-0 p-0 "
        >
          <ListItem
            className="ml-72 _dfirej mb-1"
            isSelected={false}
          >
            <li
              className="d-block position-relative ml-72 _dfirej mb-1 z-index-1"
            >
              <div
                className="ml--72 min-height-56 position-relative"
              >
                <Item
                  className={null}
                  image={null}
                  isDisabled={false}
                  isSelected={false}
                  link="/category/31"
                  onClick={null}
                  rightComponent={null}
                  title="Headline"
                >
                  <Glow
                    className={null}
                    color="#f7f7f7"
                    styles={
                      Object {
                        "container": null,
                        "glow": null,
                      }
                    }
                  >
                    <div
                      aria-hidden={true}
                      className="position-relative"
                      onTouchTap={[Function]}
                      style={null}
                    >
                      <div
                        className="position-absolute left-0 top-0 w-100P h-100P b-transparent _1dnc6ae z-index-0"
                        style={Object {}}
                      />
                      <Link
                        className=""
                        href="/category/31"
                        onClick={null}
                      >
                        <div
                          className="p-0 m-0 border-none text-align-left align-items-stretch w-100P "
                          onClick={[Function]}
                          role="link"
                        >
                          <Grid
                            className="align-items-center min-height-56 p-8px_0 position-relative z-index-2"
                            component="div"
                            wrap={false}
                          >
                            <div
                              className="align-items-center min-height-56 p-8px_0 position-relative z-index-2 d-flex min-width-100P"
                            >
                              <div
                                className="align-self-flex-start flex-shrink-0 m-0_16px w-40"
                              />
                              <GridItem
                                className="w-100P mt-2 pr-16 hyphens-auto _br4lpt word-break-break-word"
                                component="div"
                                grow={1}
                                shrink={1}
                              >
                                <div
                                  className="w-100P mt-2 pr-16 hyphens-auto _br4lpt word-break-break-word flex-grow-1"
                                >
                                  Headline
                                </div>
                              </GridItem>
                            </div>
                          </Grid>
                        </div>
                      </Link>
                    </div>
                  </Glow>
                </Item>
              </div>
            </li>
          </ListItem>
          <ListItem
            className="ml-72"
            isSelected={false}
          >
            <li
              className="d-block position-relative ml-72 z-index-1"
            >
              <div
                className="ml--72 min-height-56 position-relative"
              >
                <Item
                  className={null}
                  image={null}
                  isDisabled={false}
                  isSelected={false}
                  link="/category/32"
                  onClick={null}
                  rightComponent={null}
                  title="Headline"
                >
                  <Glow
                    className={null}
                    color="#f7f7f7"
                    styles={
                      Object {
                        "container": null,
                        "glow": null,
                      }
                    }
                  >
                    <div
                      aria-hidden={true}
                      className="position-relative"
                      onTouchTap={[Function]}
                      style={null}
                    >
                      <div
                        className="position-absolute left-0 top-0 w-100P h-100P b-transparent _1dnc6ae z-index-0"
                        style={Object {}}
                      />
                      <Link
                        className=""
                        href="/category/32"
                        onClick={null}
                      >
                        <div
                          className="p-0 m-0 border-none text-align-left align-items-stretch w-100P "
                          onClick={[Function]}
                          role="link"
                        >
                          <Grid
                            className="align-items-center min-height-56 p-8px_0 position-relative z-index-2"
                            component="div"
                            wrap={false}
                          >
                            <div
                              className="align-items-center min-height-56 p-8px_0 position-relative z-index-2 d-flex min-width-100P"
                            >
                              <div
                                className="align-self-flex-start flex-shrink-0 m-0_16px w-40"
                              />
                              <GridItem
                                className="w-100P mt-2 pr-16 hyphens-auto _br4lpt word-break-break-word"
                                component="div"
                                grow={1}
                                shrink={1}
                              >
                                <div
                                  className="w-100P mt-2 pr-16 hyphens-auto _br4lpt word-break-break-word flex-grow-1"
                                >
                                  Headline
                                </div>
                              </GridItem>
                            </div>
                          </Grid>
                        </div>
                      </Link>
                    </div>
                  </Glow>
                </Item>
              </div>
            </li>
          </ListItem>
        </ul>
      </List>
    </List>
  </div>
</CategoryListWidget>
`;
